src/EditorFeatures/Test/CommentSelection/ToggleLineCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,1084 @@
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultipleLinesPartiallyCommentedSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|//var i = 1;
+        var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        ////var i = 1;
+        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultipleLinesWithCommentsInLineSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1; // A comment.
+        var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.
+        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultipleLinesWithDifferentIndentationsSelected()
+        {
+            var markup =
+@"
+class C
+{
+[|    void M()
+    {
+        var i = 1;
+
+        var j = 2;
+    }|]
+}";
+            var expected =
+@"
+class C
+{
+[|    //void M()
+    //{
+    //    var i = 1;
+
+    //    var j = 2;
+    //}|]
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultiCaret()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var [||]i = 1;
+        var [||]j = 2;
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+[|        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultiSeletion()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1;|]
+        [|var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+[|        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultiSeletionPartiallyCommented()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|//var i = 1;|]
+        [|var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        ////var i = 1;|]
+[|        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_VisualBasicStyleComment()
+        {
```

Comment:
so all tests are in the core. don't we usually put each language-specific test in language specific test dll?

Reply:
Yes, I had the choice of adding a reference to the generic test project in csharp / vb projects or adding the VB tests here.

I also looked, there are multiple test files inside the core that do both VB and C# tests., for example http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.UnitTests/MetadataAsSource/MetadataAsSourceTests.cs,d87b0c318c53f638

Since this pattern already existed, I just continued it.